setcontext/setcontext.py

```diff
@@ -0,0 +1,406 @@
+    json_str = output.decode('utf-8')
+    json_data = json.loads(json_str)
+
+    project_names = [project['name'] for project in json_data]
+
+    return True if project_name in project_names else False
+
+
+def does_project_exist(project_name: str) -> bool:
+    """
+
+    Test if a directory exists in the local git repo.
+
+    :param project_name:
+        Name of the project.
+    :return: bool
+        True if a folder of name project_name exists in the local git repo.
+
+    """
+
+    path = PATH
+    path /= project_name
+    is_dir = path.is_dir()
+    if is_dir:
+        print(f"echo Project exists: {path};")
+    else:
+        print(f"echo Project does not exist: {path};")
+    return path.is_dir()
+
+
+def does_service_version_exist(service_or_version_name: str) -> bool:
+    """
+
+    Function to test if the service or version exists. Essentially checking if a sub namespace exists.
+
+    :param service_or_version_name:
+        Name of the service or version
+
+    :return: bool
+        True if the path is an existing directory.
+    """
+
+    path = pathlib.Path.cwd()
+    path /= service_or_version_name
+    return path.is_dir()
+
+
+def is_version_string_valid(version_string: str) -> bool:
+    """
+
+    Checks if the version string is of the right format.
+
+    :param version_string:
+        A string in the format v<XXX>
+    :return:  bool
+        True if version_string starts with v, and ends with 3 numbers.
+
+    """
+
+    #replace with regex
+    return version_string.startswith("v") and version_string[1:].isdigit() and len(version_string) == 4
+
+
+def split_namespace(namespace: str):
+    """
+
+    Splits the namespace string into seperate variables.
+
+    :param namespace:
+        The context namespace in the format project:service:version
+
+    :rtype: str,str,str
+        the project, service and version seperated into their own variables. If contexts weren't present in the namespace
+        the strings return empty.
+
+    """
+    project = ""
+    service = ""
+    version = ""
+
+    # replace with argparse for better error handling and help?
+    if not ":" in namespace:
+        project = namespace
+        service = ""
+        version = ""
+
+    else:
+        split_context = namespace.split(":")
+        split_count = len(split_context)
+
+        if split_count == 2:
+            project, service = split_context
+            version = ''
+
+        if split_count == 3:
+            project, service, version = split_context
+            is_version_string_valid(version)
+
+    return project, service, version
+
+
+def set_context_env_variable(context_type: str, value: str) -> None:
+    """
+
+    :param context_type:
+        The type of variable replaced to set context, either PROJECT, SERVICE or VERSION
+
+    :param value:
+        The value to set for the environment variable.
+
+    """
+    type_upper = context_type.upper()
+    if type_upper in ["PROJECT", "SERVICE", "VERSION"]:
```

Comment:
nit: optimization
Use tuple (immutable) instead of list. Doesn't matter with 3 elements but way faster for larger lists.